src/Plutarch/Extra/FixedDecimal.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module Plutarch.Extra.FixedDecimal (
+    PFixedDecimal (..),
+    DivideSemigroup (..),
+    DivideMonoid (..),
+    decimalToAdaValue,
+) where
+
+import Plutarch.Api.V1
+import Plutarch.Api.V1.Value
+import Plutarch.Bool
+import Plutarch.Integer
+import Plutarch.Prelude
+import Plutarch.Unsafe
+
+decimalUnit :: Term s PInteger
+decimalUnit = 1000000
```

Comment:
I don't like this being hardcoded so much - since we have plutarch we can think about doing better.
I have some ideas:

- we could have the decimal unit be a type-level Nat, which `PFixedDecimal` is tagged with. 
- then (in combination with `plutarch-safe-money`!? we could define a `Unit` typeclass, which defines the 'base' of various units, which we can then use to define typed unit conversions!

Reply:
let me whip up an example, as I think this is also adjacent to @tbidne 's time unit handling

Reply:
This sounds really nice, I'll put typelevel nat as unit decimal